debezium-connector-db2/src/main/java/io/debezium/connector/db2/Db2ConnectorConfig.java

```diff
@@ -0,0 +1,357 @@
+                RelationalDatabaseConnectorConfig.TABLE_BLACKLIST,
+                RelationalDatabaseConnectorConfig.COLUMN_BLACKLIST,
+                RelationalDatabaseConnectorConfig.SNAPSHOT_SELECT_STATEMENT_OVERRIDES_BY_TABLE,
+                RelationalDatabaseConnectorConfig.TABLE_IGNORE_BUILTIN,
+                Heartbeat.HEARTBEAT_INTERVAL, Heartbeat.HEARTBEAT_TOPICS_PREFIX,
+                CommonConnectorConfig.SOURCE_STRUCT_MAKER_VERSION
+        );
+        Field.group(config, "Connector", CommonConnectorConfig.POLL_INTERVAL_MS, CommonConnectorConfig.MAX_BATCH_SIZE,
+                CommonConnectorConfig.MAX_QUEUE_SIZE, CommonConnectorConfig.SNAPSHOT_DELAY_MS, CommonConnectorConfig.SNAPSHOT_FETCH_SIZE,
+                RelationalDatabaseConnectorConfig.DECIMAL_HANDLING_MODE, RelationalDatabaseConnectorConfig.TIME_PRECISION_MODE);
+
+        return config;
+    }
+
+    private final String databaseName;
+    private final SnapshotMode snapshotMode;
+    private final SnapshotIsolationMode snapshotIsolationMode;
+    private final ColumnNameFilter columnFilter;
+
+    public Db2ConnectorConfig(Configuration config) {
+        super(config, config.getString(SERVER_NAME), new SystemTablesPredicate(), x -> x.schema() + "." + x.table());
+
+        this.databaseName = config.getString(DATABASE_NAME);
+        this.snapshotMode = SnapshotMode.parse(config.getString(SNAPSHOT_MODE), SNAPSHOT_MODE.defaultValueAsString());
+        this.snapshotIsolationMode = SnapshotIsolationMode.parse(config.getString(SNAPSHOT_ISOLATION_MODE), SNAPSHOT_ISOLATION_MODE.defaultValueAsString());
+        this.columnFilter = getColumnNameFilter(config.getString(RelationalDatabaseConnectorConfig.COLUMN_BLACKLIST));
+    }
+
+    private static ColumnNameFilter getColumnNameFilter(String excludedColumnPatterns) {
+        return new ColumnNameFilter() {
+
+            Predicate<ColumnId> delegate = Predicates.excludes(excludedColumnPatterns, ColumnId::toString);
+
+            @Override
+            public boolean matches(String catalogName, String schemaName, String tableName, String columnName) {
+                // ignore database name as it's not relevant here
+                return delegate.test(new ColumnId(new TableId(null, schemaName, tableName), columnName));
+            }
+        };
+    }
+
+    public String getDatabaseName() {
+        return databaseName;
+    }
+
+    public SnapshotIsolationMode getSnapshotIsolationMode() {
+        return this.snapshotIsolationMode;
+    }
+
+    public SnapshotMode getSnapshotMode() {
+        return snapshotMode;
+    }
+
+    public ColumnNameFilter getColumnFilter() {
+        return columnFilter;
+    }
+
+    @Override
+    protected SourceInfoStructMaker<? extends AbstractSourceInfo> getSourceInfoStructMaker(Version version) {
+        switch (version) {
+        case V1:
+            return new LegacyV1Db2SourceInfoStructMaker(Module.name(), Module.version(), this);
```

Comment:
The legacy format shouldn't be needed for DB2. This was added to existing connectors in 0.10, as we did some backwards-incompatible changes to the `source` structure of change events. So this option was provided as means of easing migration, allowing users for keeping the old format for some time. But as you start from scratch, it's not needed really.

Reply:
Removed in the latest version.